woocommerce/woocommerce-android
# Issue/5878 filter by api version

Closes: #5878 


### Description
Due to App Store guidelines that disallow upsells for payments outside of In-App Purchase, the mobile apps cannot show all the inbox notes as in core where some notes have actions that link to pages with payment CTAs. The API work https://github.com/woocommerce/woocommerce-ios/issues/5950 has yet begun, and might take a while after our API team has bandwidth and after the merging of WC Admin to core is done (ref pbIJXs-1iI-p2).

For stores with a lower WC version, we are going to hide the Inbox menu. This PR added an eligibility use case to check the WC plugin version and only show the Inbox menu when the WC version is above a minimum version (TBD) and the feature flag is enabled.

### Testing instructions
The `INBOX_MINIMUM_SUPPORTED_VERSION` has been set to `v6.4.0` to make it easy for testing as the latest' version to this day is `6.4.1` so it should not be hard to find a site with an older version to test switching sites and see how inbox entry point is gone from More Menu. 

https://user-images.githubusercontent.com/2663464/163799684-7a93d032-ebf2-44e8-9230-f74e67a37f67.mp4
